application/event_processor/event_processor.py

```diff
@@ -237,16 +240,16 @@ def lambda_handler(event: SQSEvent, context: LambdaContext, metrics) -> None:
         return
 
     try:
-        validate_event(change_event)
-
+        service_type: ServiceType = validate_event(change_event)
+        logger.append_keys(service_type=service_type.name)
         nhs_entity = NHSEntity(change_event)
         logger.append_keys(ods_code=nhs_entity.odscode)
         logger.append_keys(org_type=nhs_entity.org_type)
         logger.append_keys(org_sub_type=nhs_entity.org_sub_type)
         metrics.set_property("ods_code", nhs_entity.odscode)
         logger.info("Created NHS Entity for processing", extra={"nhs_entity": nhs_entity})
-        event_processor = EventProcessor(nhs_entity)
-        matching_services = event_processor.get_matching_services()
+        event_processor = EventProcessor(nhs_entity, service_type)
```

Comment:
feels odd to have constructor that takes an nhs_entity and a service type , as service type can be derived from properties on the nhs entity, this would allow an consumer of this method to create an EventProcessor where the service_type differs from the real one derivable from the organisation type on the nhs_entity